### PR TITLE
https://issues.redhat.com/browse/ACM-7817 Add CIM networking

### PR DESCRIPTION
**Pull request by oafischer**

### CODE REVIEW

File: clusters/cluster_lifecycle/cim_network.adoc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+apiVersion: agent-install.openshift.io/v1beta1
+kind: NMStateConfig
+metadata:
+  name: mynmstateconfig <1>
+  namespace: <your-infraenv-namespace> <2>
+  labels:
+    some-key: <some-value> <3>
+spec:
+  config: <4>
+    interfaces: <5>
```
Comment (avishayt) [inline clusters/cluster_lifecycle/cim_network.adoc]:
```suggestion
    interfaces:
```

File: clusters/cluster_lifecycle/cim_network.adoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      config:
+        - destination: 0.0.0.0/0
+          next-hop-address: 192.168.111.1
+          next-hop-interface: eth1
+          table-id: 254
+        - destination: 0.0.0.0/0
+          next-hop-address: 192.168.140.1
+          next-hop-interface: eth1
+          table-id: 254
+  interfaces:
```
Comment (avishayt) [inline clusters/cluster_lifecycle/cim_network.adoc]:
```suggestion
  interfaces: <5>
```

**Review by avishayt — COMMENTED** [review]
Looks great, just the <5> is on the wrong line

File: clusters/cluster_lifecycle/cim_network.adoc [hunk 3]
```diff
@@ -0,0 +1,97 @@
+[#cim-network-config]
+= Configuring advanced networking for an infrastructure environment
+
+For hosts that require networking beyond DHCP on a single interface, you must configure advanced networking. The required configuration includes creating one or more instances of the `NMStateConfig` resource that describes the networking for one or more hosts.
```
Comment (dockerymick) [inline clusters/cluster_lifecycle/cim_network.adoc:4]: What's DHCP?

**Review by dockerymick — APPROVED** [review]
Great job with adding this content, Oliver. I also checked the links and they are working. Left a question, but this looks good to me

File: clusters/cluster_lifecycle/cim_network.adoc (hunk 3 above)
Comment (avishayt) [reply clusters/cluster_lifecycle/cim_network.adoc:4]: Dynamic Host Configuration Protocol, though it's one of those acronyms that more people know the letters and not what it spells out, like DNA or IBM.  Anyone configuring cluster networking should be familiar with DHCP.

Comment (oafischer) [reply clusters/cluster_lifecycle/cim_network.adoc:4]: I would tend to agree, I find DHCP more recognizable than spelling it out. If everything else looks good I'll go ahead and merge

**Pull request merged**